### PR TITLE
small optimization for one call value

### PR DESCRIPTION
**Pull request by Kobenko**

### CODE REVIEW

**Comment by Kobenko** [comment]
@densh Do you need this refactoring? and is it worth to contribute to the version earlier than 2.13?

File: src/compiler/scala/reflect/quasiquotes/Parsers.scala [hunk 1]
```diff
@@ -153,9 +152,8 @@ trait Parsers { self: Quasiquotes =>
 
       override def refineStat(): List[Tree] =
         if (isHole && !isDclIntro) {
-          val result = RefineStatPlaceholder(in.name) :: Nil
           in.nextToken()
-          result
+          return RefineStatPlaceholder(in.name) :: Nil
```
Comment (lrytz) [inline src/compiler/scala/reflect/quasiquotes/Parsers.scala]: @Kobenko what is the motivation behind this change? How can we know that `in.nextToken()` doesn't change `in.name`?

Comment (Kobenko) [reply src/compiler/scala/reflect/quasiquotes/Parsers.scala]: @lrytz you are right, this change (as well as changes below) is misleading, if you don't know what `nextToken()` is a side effect. Return back

**Comment by lrytz** [comment]
OK, now the remaining change is cosmetic, it just saves a local variable. Should this PR remain open?

**Comment by Kobenko** [comment]
@lrytz If in your opinion this cosmetic solution is not worthy of a merger, you can close it, and I can solve more serious things. Let's just say, this PR was as a research, that a simple person can help the community

**Comment by lrytz** [comment]
> Let's just say, this PR was as a research, that a simple person can help the community

We're really thankful for the countless contributions we get from our community, so please feel welcome to join!

I'm going to close this PR, as it doesn't fix any problem.
